commerce_coordinator/apps/titan/clients.py

```diff
@@ -0,0 +1,103 @@
+"""
+API clients for Titan.
+"""
+
+import requests
+from celery.utils.log import get_task_logger
+from django.conf import settings
+from edx_rest_api_client.client import OAuthAPIClient
+
+# Use special Celery logger for tasks client calls.
+logger = get_task_logger(__name__)
+
+
+class TitanAPIClient():
+    """
+    API client for calls to Titan using API key.
+    """
+
+    def __init__(self):
+        self.client = requests.Session()
+        # Always send API key.
+        self.client.headers.update(self.api_key_header)
+
+    @property
+    def api_base_url(self):
+        """URL of API service."""
+        return str(settings.TITAN_URL).strip('/') + '/v1'
+
+    @property
+    def api_key_header(self):
+        """Header to add to all API requests to authenticate to endpoint."""
+        return {'X-API-Key': settings.TITAN_API_KEY}
+
+    @property
+    def normal_timeout(self):
+        """
+        Shortcut for a normal timeout. Must be manually applied to each request.
+
+        See https://requests.readthedocs.io/en/latest/user/quickstart/#timeouts.
+        """
+        return (
+            settings.REQUEST_CONNECT_TIMEOUT_SECONDS,
+            settings.REQUEST_READ_TIMEOUT_SECONDS
+        )
+
+    def post(self, resource_path, data):
+        """
+        Send a POST request to a Titan API resource.
+
+        Arguments:
+            resource_path: the path of the API resource to POST to
+            data: the dictionary to send to the API resource
+        Returns:
+            dict: Dictionary represention of JSON returned from API
+
+        """
+        try:
+            resource_url = self.api_base_url.strip('/') + resource_path
```

Comment:
See my prior question about this pattern for building urls.  `strip()` here is not technically needed, and is inconsistent with leaving it out on [Line 33](https://github.com/edx/commerce-coordinator/pull/46/files#diff-9e9e16f53a74ae158cfc7a9e805b26c53fb3ea088cbb1cbeb230f3c14b8f3c91R33).  Not a major issue, but it does seem like there could be a better way...

Reply:
You're right. I should be doing `strip()` where we receive user input and not litter it everywhere in the code. In this case, it should only be when reading in `settings.TITAN_URL`, because historically we've had trouble standardizing trailing slashes in environmental variables.

My original rational was to remind the coder that `api_base_url` has its trailing slash stripped, and thus `resource_path` must begin with a slash. But that is better done with an explicit check.

I'll remove the `strip()` here, assert `resource_path` begins with a slash, and add it to my list of things to add environmental variable validation for.

Reply:
Done in https://github.com/edx/commerce-coordinator/commit/d7bd1057c2260f7005fdf1a161244a60e6494cc9.